### DISCUSSION TITLE
Best way to handle the route [...slug]/contact/page.tsx

### DISCUSSION THREAD

**Discussion by CodyBontecou** [vercel/next.js#61237] [Help]
### Summary

I'd like to have a dynamic route ([...slug]) with static segments after it (/contact). Using the directory structure `[...slug]/contact/page.tsx` gives me the error: `Error: Catch-all must be the last part of the URL`.

What is the recommended way of approaching this problem? I could extract the contact param from the URL slug within `[...slug]/page.tsx` but that will become unwieldy as I have to manage multiple static segments in the last part of the URL.

### Additional information

This is using v14 and the App Router.

**Comment by ryozm** [comment]
Maybe this will be useful for you.
Using the directory structure`[[...slug]]/page.tsx`
```TypeScript
// page.tsx
const Page = ({ params }: { params: { slug: string[] } }) => {
  const slugPath = params.slug.join('/')
  if (slugPath.endsWith('/contact')) {
    // Render contact page
    return <div>Contact Page</div>
  }
  // Other
  return <div>My Post: {params.slug.join('/')}</div>
}

export default Page
```

**Comment by CodyBontecou** [comment]
I needed to not spread the slug and instead use `/[slug]/contact/page.tsx`
